src/main/java/com/leaf/billing/utils/PropertyBuilder.java

```diff
@@ -15,18 +21,70 @@ private PropertyBuilder() {
                 .collect(Collectors.toMap(keyValue -> keyValue[0], keyValue -> keyValue[1]));
+
+        if(!nonePathIsNull(sparkProperties.get(SOURCE_BASE_PATH), sparkProperties.get(TARGET_BASE_PATH))){
+            System.exit(1);
+        }
+        String normalizedPathSource = normalizePath(sparkProperties.get(SOURCE_BASE_PATH));
+        String normalizedPathTarget = normalizePath(sparkProperties.get(TARGET_BASE_PATH));
+        validateEntriesPath(normalizedPathSource, normalizedPathTarget);
         JobProperties jobProperties = new JobProperties();
-        jobProperties.setSourceBasePath(sparkProperties.get(SOURCE_BASE_PATH));
-        jobProperties.setTargetBasePath(sparkProperties.get(TARGET_BASE_PATH));
+        jobProperties.setSourceBasePath(normalizedPathSource);
+        jobProperties.setTargetBasePath(normalizedPathTarget);
         jobProperties.setSourceDeleteFiles(Boolean.valueOf(sparkProperties.getOrDefault(SOURCE_DELETE_FILES, "False")));
         return jobProperties;
     }
 
+    public static boolean nonePathIsNull(String source, String target){
+        return !(source == null || target == null);
+    }
+
+    private static void validateEntriesPath(String normalizedPathSource, String normalizedPathTarget)
+            throws URISyntaxException, IOException {
+        URI sourceURIFile = new URI(normalizedPathSource);
+        URI targetURIFile = new URI(normalizedPathTarget);
+        FileSystem sourceFileSystem = getFileSystemForFiles(sourceURIFile);
+        FileSystem targetFileSystem = getFileSystemForFiles(targetURIFile);
+        Path sourcePathFile = getPathForFiles(sourceURIFile);
+        Path targetPathFile = getPathForFiles(targetURIFile);
+        if(pathExist(sourceFileSystem, sourcePathFile) && pathExist(targetFileSystem, targetPathFile)){
+            if(notIsADirectory(sourceFileSystem, sourcePathFile) || notIsADirectory(targetFileSystem, targetPathFile)){
+                System.exit(1);
+            }
+        }else{
+            System.exit(1);
+        }
+    }
+
+    private static FileSystem getFileSystemForFiles(URI fileUri) throws IOException {
+        Configuration conf = new Configuration();
+        return org.apache.hadoop.fs.FileSystem.get(fileUri, conf);
+    }
+
+    private static Path getPathForFiles(URI fileUri) {
+        return new org.apache.hadoop.fs.Path(fileUri);
+    }
+
+    public static boolean pathExist(FileSystem fileSystem, Path folderPath) throws IOException {
+        return fileSystem.exists(folderPath);
+    }
+
+    public static boolean notIsADirectory(FileSystem fileSystem, Path folderPath) throws IOException {
+        return !(fileSystem.getFileStatus(folderPath).isDirectory());
+    }
+
+    private static String normalizePath(String path){
+        String normalizedPath = path;
+        if(path.length()-1 == path.lastIndexOf("/")){
+            normalizedPath = path.substring(0,path.length()-1);
+        }
+        return normalizedPath;
+    }
```

Comment:
Is there a trim method in Java 
```suggestion
    private static String normalizePath(String path){
        return path.endsWith("/") ? StringUtils.chop(path) : path;
    }
```
I don't think I've ever used StringUtils but I saw it being used in one of Fabyan's PR, you could take a look into it to see if something like what I'm suggesting is possible. I only did a quick google search for chop and endsWith, maybe there's a `trim` method that already does exactly this :man_shrugging:

Reply:
in case you're not familiar with ternary, it works like this: (condition) ? (value if condition true) : (value if condition false)

Reply:
Thanks, Ricardo! I searched in StringUtils and found the strip method that solves all things in one line, then here is the commit that implemented it: https://github.com/Leaf-Agriculture/billing-data-aggregation-job/pull/6/commits/4bb4f8a8928a3e5eef6c275df2209258a069cac1

Reply:
nice, does this keep the slash if its in the beginning of the path? maybe we should use stripEnd :thinking:

Reply:
Thanks Ricardo, I changed it on this commit: https://github.com/Leaf-Agriculture/billing-data-aggregation-job/pull/6/commits/ff524cba7f9e770d8e74999ec97e9d78bb033e72